tools/booster/Proxy.hs

```diff
@@ -324,57 +336,69 @@ respondEither mbStatsVar booster kore req = case req of
         simplifyResult res@ExecuteResult{reason, state, nextStates} = do
             Log.logInfoNS "proxy" . Text.pack $ "Simplifying state in " <> show reason <> " result"
-            (simplifiedState, simplifiedStateLogs) <- simplifyExecuteState logSettings mbModule state
-            simplifiedNexts <- maybe (pure []) (mapM $ simplifyExecuteState logSettings mbModule) nextStates
-            let (filteredNexts, filteredNextsLogs) = unzip . filter (not . isBottom . fst) $ simplifiedNexts
-            let result = case reason of
-                    Branching
-                        | null filteredNexts ->
-                            res{reason = Stuck, nextStates = Nothing}
-                        | length filteredNexts == 1 ->
-                            res -- What now? would have to re-loop. Return as-is.
-                            -- otherwise falling through to _otherReason
-                    CutPointRule
-                        | null filteredNexts ->
-                            -- HACK. Would want to return the prior state
-                            res{reason = Stuck, nextStates = Nothing}
-                    _otherReason ->
-                        res
-                            { state = simplifiedState
-                            , nextStates = if null filteredNexts then Nothing else Just filteredNexts
-                            }
-            let allLogs =
-                    if null filteredNexts
-                        then simplifiedStateLogs
-                        else combineLogs $ simplifiedStateLogs : filteredNextsLogs
-            pure $ appendLogs result allLogs
+            simplified <- simplifyExecuteState logSettings mbModule state
 
-        isBottom :: ExecuteState -> Bool
-        isBottom ExecuteState{term}
-            | KoreJson.KJBottom _ <- term.term = True
-        isBottom ExecuteState{predicate = Just p}
-            | KoreJson.KJBottom _ <- p.term = True
-        isBottom _ = False
+            case simplified of
+                Left logsOnly -> do
+                    -- state simplified to \bottom, return vacuous
+                    Log.logInfoNS "proxy" "Vacuous after simplifying result state"
+                    pure $ makeVacuous logsOnly res
+                Right (simplifiedState, simplifiedStateLogs) -> do
+                    simplifiedNexts <-
+                        maybe
+                            (pure [])
+                            (mapM $ simplifyExecuteState logSettings mbModule)
+                            nextStates
+                    let (logsOnly, (filteredNexts, filteredNextLogs)) =
+                            second unzip $ partitionEithers simplifiedNexts
+                        newLogs = simplifiedStateLogs : logsOnly <> filteredNextLogs
+
+                    pure $ case reason of
+                        Branching
+                            | null filteredNexts ->
+                                res{reason = Stuck, nextStates = Nothing}
```

Comment:
I think we're losing logs again here? Since we return the result from every brunch of the case expression, we need to include logs here to.
```suggestion
                                res{reason = Stuck, nextStates = Nothing, logs = combineLogs $ res.logs : simplifiedStateLogs : logsOnly}
```